.circleci/config.yml

```diff
@@ -1,28 +1,71 @@
-version: 2
-jobs:
-  build:
+version: 2.1
+commands:
+  runtests:
+    steps:
+      - checkout
+      - restore_cache: # Download and cache dependencies
+          key: libhoney-java-{{ checksum "pom.xml" }}
+      - run: ./mvnw dependency:go-offline # gets the project dependencies
+      - save_cache: # saves the project dependencies
+          paths:
+            - ~/.m2
+          key: libhoney-java-{{ checksum "pom.xml" }}
+      - run: ./mvnw package # run the actual tests
+executors:
+  java8:
```

Comment:
These executors feel very close to being the same except for the version of java. Close enough that it may be more clear to send the version of java to use as a parameter to the executor instead of defining each one separately. As an example of using one executor and parameters to set the version, I can recommend looking at https://github.com/honeycombio/libhoney-go/blob/c8c127c0f72621575cd02c94b07705a3585f263c/.circleci/config.yml

Reply:
You are 100% correct and I have implemented this. Thanks for the reference, as I am still learning CircleCI! I have updated the PR with your suggestion.